com.ibm.wala.core.tests/src/com/ibm/wala/core/tests/ir/AnnotationTest.java

```diff
@@ -37,7 +37,7 @@
 import java.util.Set;
 import org.junit.Test;
 
-public class AnnotationTest extends WalaTestCase {
+public abstract class AnnotationTest extends WalaTestCase {
```

Comment:
Why make this (and other test classes) abstract?

Reply:
These specific test classes cannot be instantiated by typical JUnit test runners. These are apparently intended as base classes for real tests. IntelliJ IDEA 2019.2 adds an inspection for this, called "Unconstructable JUnit TestCase":

> Reports non-abstract JUnit test cases which do not expose a public no-arg constructor or a public constructor which takes a single string as an argument. Such test cases will be unrunnable by most JUnit test runners, including IDEA's.

Declaring such classes `abstract` resolves the inspection warnings and IMHO accurately reflects the intent that these classes only be used as base classes, never instantiated directly.

Reply:
👍